tests/Model/IndexInfoFunctionalTest.php

```diff
@@ -0,0 +1,153 @@
+        $result->next();
+
+        $this->assertEquals(5, $result->current()->getBucketSize());
+    }
+
+    public function testGetDefaultLanguage()
+    {
+        $this->collection->createIndex(['x' => 'text']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        $this->assertEquals('english', $result->current()->getDefaultLanguage());
+    }
+
+    public function testGetLanguageOverride()
+    {
+        $this->collection->createIndex(['x' => 'text']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        $this->assertEquals('language', $result->current()->getLanguageOverride());
+    }
+
+    public function testGetPartialFilterExpression()
+    {
+        $this->collection->createIndex(['x' => 1], ['partialFilterExpression' => ['x' => ['$gt' => 5]]]);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        $this->assertSameDocument(['x' => ['$gt' => 5]], $result->current()->getPartialFilterExpression());
+    }
+
+    public function testGetStorageEngine()
+    {
+        $this->collection->createIndex(
+                        ['x' => 1],
+                        [
+                            'storageEngine' => ['wiredTiger' => ['configString' => '']]
+                        ]
+                );
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        $this->assertSameDocument(['wiredTiger' => ['configString' => '']], $result->current()->getStorageEngine());
+    }
+
+    public function testGetTextIndexVersion()
+    {
+        $this->collection->createIndex(['x' => 'text']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        if (version_compare($this->getServerVersion(), '3.2.0', '<')) {
+            $this->assertEquals(2, $result->current()->getTextIndexVersion());
+            return;
+        }
+        $this->assertEquals(3, $result->current()->getTextIndexVersion());
+    }
+
+    public function testGetWeights()
+    {
+        $this->collection->createIndex(['x' => 'text']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+
+        $this->assertSameDocument(['x' => 1], $result->current()->getWeights());
```

Comment:
I would probably add one more test here that actually sets up the weights for different text fields, as is described [here](https://docs.mongodb.com/manual/tutorial/control-results-of-text-search/).